lua/neo-tree/sources/common/components.lua

```diff
@@ -147,4 +147,41 @@ M.name = function(config, node, state)
   }
 end
 
+local prevent_indent = {}
+
+M.indent = function(config, node, state)
+  local indent_size = config.indent_size or 2
+  local level = node.level
+  local with_markers = config.with_markers
+
+  if indent_size == 0 or level < 2 or not with_markers then
+    return { text = string.rep(" ", indent_size * level) }
+  end
+
+  local indent_marker = config.indent_marker or "│"
+  local last_indent_marker = config.last_indent_marker or "└"
+  local highlight = config.highlight or highlights.INDENT_MARKER
+
+  prevent_indent[level] = node.is_last_child
```

Comment:
Also, should this be named `skip_marker_at_level` instead? Is that what it is actually recording? As far as I can see, it is only being set once and it is entirely determined by whether `i == level and node.is_last_child`. If that is the case, there's no reason to have it all. it could be a local boolean variable or just insert the condition itself.

Reply:
I think `skip_marker_at_level` sounds better, I'll rename the variable. Also, I will put `skip_marker_at_level` in the state.

What about making the variable local. The previous level may affect the following levels. So if I use a local variable, I have the following problem:
<img width="259" alt="problem" src="https://user-images.githubusercontent.com/57654917/152231702-e3e825d1-a786-40a1-8dd1-363f2682ee57.png">

How it should be and how it works with a non-local variable:
<img width="259" alt="normal" src="https://user-images.githubusercontent.com/57654917/152231960-bf4a6201-fbd4-4fe9-b9a4-e6f4b12dc6e0.png">